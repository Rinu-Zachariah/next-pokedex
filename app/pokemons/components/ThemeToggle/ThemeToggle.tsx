import React, { useEffect, useState } from 'react';
import { useTheme } from "next-themes";

const ThemeToggle: React.FC = () => {
  const { theme, setTheme } = useTheme();
  const [mounted, setMounted] = useState(false);

  useEffect(() => setMounted(true), []);
  if (!mounted) return null;
  
  return ( 
    <button
      className="p-2 rounded border focus:outline focus:ring"
      onClick={() => setTheme(theme === "dark" ? "light" : "dark")}
      aria-label="Toggle theme"
    >
      {theme === "dark" ? "Dark" : "Light"}
    </button>
  );
};

export default ThemeToggle;