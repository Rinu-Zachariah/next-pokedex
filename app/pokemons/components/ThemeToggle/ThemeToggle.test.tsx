import React from 'react';
import { render } from '@testing-library/react';
import ThemeToggle from './ThemeToggle';

describe('ThemeToggle', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<ThemeToggle />);
    expect(asFragment()).toMatchSnapshot();
  });
});
