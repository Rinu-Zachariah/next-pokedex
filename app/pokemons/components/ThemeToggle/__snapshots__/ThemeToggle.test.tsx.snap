// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeToggle renders correctly 1`] = `
<DocumentFragment>
  <button
    aria-label="Toggle theme"
    class="p-2 rounded border focus:outline focus:ring"
  >
    Light
  </button>
</DocumentFragment>
`;
